pkg/controller/constraint/constraint_controller.go

```diff
@@ -130,8 +128,7 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 		// Handle deletion
 		if containsString(finalizerName, instance.GetFinalizers()) {
 			if _, err := r.opa.RemoveConstraint(context.Background(), instance); err != nil {
-				// TODO make the unrecognized constraint error detectable via class introspection
-				if !(strings.HasPrefix(err.Error(), "Constraint kind ") && strings.HasSuffix(err.Error(), " is not recognized")) {
+				if _, ok := err.(*opa.UnrecognizedConstraintError); ok {
```

Comment:
Currently opa.UnrecognizedConstraintError is a value, not a pointer, so this type check would never be true.

We should actually update the frameworks library to use a pointer-to-a-string, much like:

https://golang.org/src/errors/errors.go